api/envoy/extensions/filters/http/header_to_metadata/v3/header_to_metadata.proto

```diff
@@ -90,26 +90,33 @@ message Config {
     option (udpa.annotations.versioning).previous_message_type =
         "envoy.config.filter.http.header_to_metadata.v2.Config.Rule";
 
-    // The header that triggers this rule — required.
-    string header = 1
-        [(validate.rules).string = {min_bytes: 1 well_known_regex: HTTP_HEADER_NAME strict: false}];
+    // Specifies one of the header or cookie match will be performed on the value of the header.
+    oneof header_cookie_specifier {
```

Comment:
Couple things here:

1. We cannot change the numbering of existing fields or else we break wire-compatibility of existing control planes. It is ok for fields to be out-of-order with respect to their identifiers, so `cookie` can be 5.
2. We also cannot promote fields to be part of a `oneof`. So the way this can work for v3 is to remove the oneof and tag the two fields with `[(udpa.annotations.field_migrate).oneof_promotion = "header_cookie_specifier"]` and remove the `min_bytes: 1` validateion. Then in the code handles the Rule, we'll have to manually enforce that exactly one of those fields is non-empty. In v4 this will become a oneof automatically and when we convert Envoy to use the v4 protos we can reinstate the validation and remove validation code from the Rule handling.